src/main/java/com/synopsys/integration/alert/provider/blackduck/factories/BlackDuckTaskFactory.java

```diff
@@ -0,0 +1,69 @@
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.provider.blackduck.factories;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.persistence.accessor.ConfigurationAccessor;
+import com.synopsys.integration.alert.common.persistence.accessor.NotificationManager;
+import com.synopsys.integration.alert.common.persistence.accessor.ProviderDataAccessor;
+import com.synopsys.integration.alert.common.persistence.util.FilePersistenceUtil;
+import com.synopsys.integration.alert.common.provider.ProviderProperties;
+import com.synopsys.integration.alert.common.provider.lifecycle.ProviderTask;
+import com.synopsys.integration.alert.common.provider.lifecycle.ProviderTaskFactory;
+import com.synopsys.integration.alert.provider.blackduck.BlackDuckProviderKey;
+import com.synopsys.integration.alert.provider.blackduck.tasks.BlackDuckAccumulator;
+import com.synopsys.integration.alert.provider.blackduck.tasks.BlackDuckDataSyncTask;
+
+@Component
+public class BlackDuckTaskFactory implements ProviderTaskFactory {
+    private final BlackDuckProviderKey blackDuckProviderKey;
+    private final TaskScheduler taskScheduler;
+    private final ProviderDataAccessor blackDuckDataAccessor;
+    private final ConfigurationAccessor configurationAccessor;
+    private final NotificationManager notificationManager;
+    private final FilePersistenceUtil filePersistenceUtil;
+
+    @Autowired
+    public BlackDuckTaskFactory(BlackDuckProviderKey blackDuckProviderKey, TaskScheduler taskScheduler, ProviderDataAccessor blackDuckDataAccessor,
+        ConfigurationAccessor configurationAccessor, NotificationManager notificationManager, FilePersistenceUtil filePersistenceUtil) {
+        this.blackDuckProviderKey = blackDuckProviderKey;
+        this.taskScheduler = taskScheduler;
+        this.blackDuckDataAccessor = blackDuckDataAccessor;
+        this.configurationAccessor = configurationAccessor;
+        this.notificationManager = notificationManager;
+        this.filePersistenceUtil = filePersistenceUtil;
+    }
+
+    @Override
+    public List<ProviderTask> createTasks(ProviderProperties providerProperties) {
+        BlackDuckAccumulator accumulator = new BlackDuckAccumulator(blackDuckProviderKey, taskScheduler, notificationManager, filePersistenceUtil);
+        accumulator.setProviderPropertiesForRun(providerProperties);
```

Comment:
We want to get rid of the getter/setter for properties and pass them in through the constructor.

Reply:
That is part of the reason why it can't be a component.